flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractStateBackend.java

```diff
@@ -66,4 +66,9 @@ public OperatorStateBackend createOperatorStateBackend(
 			String operatorIdentifier) throws Exception {
 		return new DefaultOperatorStateBackend(env.getUserClassLoader());
 	}
+
+	@Override
+	public void close() throws Exception {
+
+	}
```

Comment:
The reason for not renaming it is because it's used in the (admittedly `@PublicEvolcing`) method `StreamExecutionEnvironment.setStateBackend()`. If we didn't have that I would also prefer renaming this to something like `StateBackendFactory`.

The fact that it is in fact a factory (that should be stateless) makes me a bit uneasy about adding a `close()` method to the `AbstractStateBackend`. Why is it needed, in fact?